opensearch-project/OpenSearch-Dashboards
# [Workspace]Show add collaborators modal until collaborators added

### Description


This PR is for updating a UX behavior when adding collaborators. The add collaborators modal will be displayed until collaborators added. The "Add collaborators" button will become disabled & loading state during the updating API call.


## Screenshot


![image](https://github.com/user-attachments/assets/ce672cf0-bbe5-4577-be20-4e899b2be4dc)


## Testing the changes



- Clone branch code and run `yarn osd bootstrap --single-version loose`
- Add below configs in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
opensearchDashboards.dashboardAdmin.users: ['admin']
```
- Run `yarn start --no-base-path`
- Login with admin user and go to an existing workspace setting page
- Click "Collaborators" tab
- Click "Add Collaborators" button
- Click "Add users" button in the dropdown
- Enter user id and click "Add Collaborators" button, the "Add Collaborators" button should be disabled & loading state 

## Changelog

- feat: [Workspace]Show add collaborators modal until collaborators added

### Check List
